golang/Chapter04/model/client.go

```diff
@@ -24,8 +25,7 @@ func (c *Client) ListItem(itemid, sellerid string, price float64) bool {
 		err := c.Conn.Watch(func(tx *redis.Tx) error {
 			if _, err := tx.TxPipelined(func(pipeliner redis.Pipeliner) error {
 				if !tx.SIsMember(inventory, itemid).Val() {
-					tx.Unwatch(inventory)
-					return nil
```

Comment:
Don't we need to unwatch?

Reply:
Yep, refer to the source code inside [redis/tx.go at v7.0.0-beta.5 · go-redis/redis](https://github.com/go-redis/redis/blob/v7.0.0-beta.5/tx.go) Line 66-92, we could know `WatchContext()` method always call `tx.Close()`  to perform UNWATCH operation before return.

Reply:
We can demonstrate that by running the unit test [redis-in-action/redisConn_test.go at golang · 1nFrastr/redis-in-action](https://github.com/1nFrastr/redis-in-action/blob/golang/golang/redisConn/redisConn_test.go#L24-L42) while redis-cli is on MONITOR mode. 

www@ubu-server:~$ redis-cli monitor
OK
1621740375.030173 [0 10.0.2.2:13139] "select" "15"
1621740375.030620 [15 10.0.2.2:13139] "ping"
1621740375.030995 [15 10.0.2.2:13139] "info" "server"
1621740375.031619 [15 10.0.2.2:13139] "watch" "key"
1621740375.334649 [0 10.0.2.2:13142] "select" "15"
1621740375.335134 [15 10.0.2.2:13142] "incr" "key"
1621740375.335644 [15 10.0.2.2:13139] "MULTI"
1621740375.335654 [15 10.0.2.2:13139] "EXEC"
1621740375.336050 [15 10.0.2.2:13139] "unwatch"
1621740375.336449 [15 10.0.2.2:13139] "get" "key"
1621740375.336848 [15 10.0.2.2:13139] "flushdb"